packages/template-retail-react-app/app/commerce-hooks/context/shopper-basket-context.js

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useEffect} from 'react'
+import {useCommerceAPI} from '../../commerce-api/contexts'
+import useCustomer from '../../commerce-api/hooks/useCustomer'
+import basketsReducer from '../reducers/baskets-reducer'
+
+const basketsInitialState = {
+    isLoading: false,
+    error: false,
+    data: [],
+    total: 0
+}
+const BasketsContext = React.createContext()
+const BasketsProvider = ({children}) => {
+    // TODO: we should separate api into different apis context instead of having a combined api object here
+    // for example: ShopperBasketAPI, ShopperCustomerAPI
+    // expect to use const api = shopperBasketAPI()
+    const api = useCommerceAPI()
```

Comment:
TODO: In a real implementation, we should be able to have separate API services for each endpoint. For example: shopperBasketsAPI(), shopperCustomersAPI(). Here I used the combined api service that available in the current implementation temporarily only to demonstrate the flow of state management